stm32/lib/controller/src/communication.h

```diff
@@ -0,0 +1,63 @@
+#ifndef STM32_LIB_CONTROLLER_SRC_COMMUNICATION_H_
+#define STM32_LIB_CONTROLLER_SRC_COMMUNICATION_H_
+
+#include <stdint.h>
+
+#include "soil_power_sensor.pb.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef enum {
+  /** Success */
+  CONTROLLER_SUCCESS = 0,
+  /** General error */
+  CONTROLLER_ERROR,
+  /** Timeout for i2c functions */
+  CONTROLLER_TIMEOUT,
+  /** For mutex violations */
+  CONTROLLER_MUTEX
+} ControllerStatus;
+
+typedef struct {
+  /** State in buffer */
+  uint8_t* data;
+  /** Size of buffer */
+  size_t size;
+  /** Length of data in buffer */
+  size_t len;
+} Buffer;
+
+/** @brief Buffer for ControllerTransmit */
+static Buffer tx = {};
```

Comment:
This construct makes a Buffer variable per translation unit (i.e. file) that includes this header. Is this intentional? Or is the intent to have global `tx` and `rx` buffers?

Additionally, in C I think `= {0}` is needed to zero initialize stuff.